capsules/src/net/thread/tlv.rs

```diff
@@ -0,0 +1,1315 @@
+                let (offset, link_quality_2) = dec_try!(buf, offset; decode_u8);
+                let (offset, link_quality_1) = dec_try!(buf, offset; decode_u8);
+                let (offset, leader_cost) = dec_try!(buf, offset; decode_u8);
+                let (offset, id_sequence) = dec_try!(buf, offset; decode_u8);
+                let (offset, active_routers) = dec_try!(buf, offset; decode_u8);
+                let mut offset = offset;
+                let mut sed_buffer_size = None;
+                if offset + mem::size_of::<u16>() < length as usize {
+                    let (new_offset, sed_buffer_size_raw) = dec_try!(buf, offset; decode_u16);
+                    offset = new_offset;
+                    sed_buffer_size = Some(sed_buffer_size_raw);
+                }
+                let mut sed_datagram_count = None;
+                if offset + mem::size_of::<u8>() < length as usize {
+                    let (new_offset, sed_datagram_count_raw) = dec_try!(buf, offset; decode_u8);
+                    offset = new_offset;
+                    sed_datagram_count = Some(sed_datagram_count_raw);
+                }
+                stream_done!(offset,
+                             Tlv::Connectivity {
+                                 parent_priority: parent_priority,
+                                 link_quality_3: link_quality_3,
+                                 link_quality_2: link_quality_2,
+                                 link_quality_1: link_quality_1,
+                                 leader_cost: leader_cost,
+                                 id_sequence: id_sequence,
+                                 active_routers: active_routers,
+                                 sed_buffer_size: sed_buffer_size,
+                                 sed_datagram_count: sed_datagram_count,
+                             })
+            }
+            TlvType::LinkMargin => {
+                let (offset, link_margin) = dec_try!(buf, offset; decode_u8);
+                stream_done!(offset, Tlv::LinkMargin(link_margin))
+            }
+            TlvType::Status => {
+                let (offset, status) = dec_try!(buf, offset; decode_u8);
+                stream_done!(offset, Tlv::Status(status))
+            }
+            TlvType::Version => {
+                let (offset, version) = dec_try!(buf, offset; decode_u16);
+                stream_done!(offset, Tlv::Version(version))
+            }
+            TlvType::ActiveOperationalDataset => {
+                stream_done!(offset + length as usize,
+                             Tlv::ActiveOperationalDataset(&buf[offset..offset + length as usize]))
+            }
+            TlvType::PendingOperationalDataset => {
+                stream_done!(offset + length as usize,
+                             Tlv::PendingOperationalDataset(&buf[offset..offset + length as usize]))
+            }
+            TlvType::NotPresent => stream_err!(),
+        }
+    }
+}
+
+#[repr(u8)]
+pub enum TlvType {
```

Comment:
I think variants are implicitly public. Tried to make them pub and got an error.